tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,200 @@
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Operator", Ordered, func() {
+	SetDefaultEventuallyTimeout(120 * time.Second)
+	SetDefaultEventuallyPollingInterval(time.Second)
+	var (
+		resourceAvailable = resourcecondition.Condition{
+			Type:   "Available",
+			Status: "True",
+		}
+		resourceReconcilied = resourcecondition.Condition{
+			Type:   "Reconciled",
+			Status: "True",
+		}
+
+		resourceReady = resourcecondition.Condition{
+			Type:   "Ready",
+			Status: "True",
+		}
+
+		istioResources = []string{
+			// TODO: Find an alternative to this list
+			"authorizationpolicies.security.istio.io",
+			"destinationrules.networking.istio.io",
+			"envoyfilters.networking.istio.io",
+			"gateways.networking.istio.io",
+			"istiorevisions.operator.istio.io",
+			"istios.operator.istio.io",
+			"peerauthentications.security.istio.io",
+			"proxyconfigs.networking.istio.io",
+			"requestauthentications.security.istio.io",
+			"serviceentries.networking.istio.io",
+			"sidecars.networking.istio.io",
+			"telemetries.telemetry.istio.io",
+			"virtualservices.networking.istio.io",
+			"wasmplugins.extensions.istio.io",
+			"workloadentries.networking.istio.io",
+			"workloadgroups.networking.istio.io",
+		}
+	)
+
+	Describe("installation", func() {
+		// TODO: we  need to support testing both types of deployment for the operator, helm and olm via subscription.
+		// Discuss with the team if we should add a flag to the test to enable the olm deployment and don't do that deployment in different step
+		When("default helm manifest are applied", func() {
+			BeforeAll(func() {
+				if skipDeploy == "true" {
+					Skip("Skipping the deployment of the operator and the tests")
+				}
+				Success("Deploying Operator using default helm charts located in /chart folder")
+				Eventually(deployOperator).Should(Succeed(), "Operator failed to be deployed; unexpected error")
+			})
+
+			It("starts successfully", func() {
+				Eventually(kubectl.GetResourceCondition).
+					WithArguments(namespace, "deployment", deploymentName).
+					Should(ContainElement(resourceAvailable))
+				Success("Operator deployment is Available")
+
+				Expect(kubectl.GetPodPhase(namespace, "control-plane=istio-operator")).Should(Equal("Running"), "Operator failed to start; unexpected pod Phase")
+				Success("Istio-operator pod is Running")
+			})
+
+			It("deploys all the CRDs", func() {
+				Eventually(kubectl.GetCRDs).
+					Should(ContainElements(istioResources), "Istio CRDs are not present; expected list to contain all elements")
+				Success("Istio CRDs are present")
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio resource", func() {
+		AfterAll(func() {
+			// Cleanup the operator at the end of the test
+			Eventually(undeployOperator).Should(Succeed(), "Operator failed to be deleted; unexpected error")
+			Success("Operator deployment is deleted")
+		})
```

Comment:
This doesn't belong under "installation and uninstallation of the istio resource". It should be one level higher (and at the end).